src/main/java/uk/gov/hmcts/reform/translate/service/DictionaryService.java

```diff
@@ -38,4 +52,46 @@ public Map<String, String> getDictionaryContents() {
 
         return Collections.emptyMap();
     }
+
+
+    public void putDictionary(final DictionaryRequest dictionaryRequest) {
+
+        val currentUser = securityUtils.getUserInfo();
+        dictionaryRequest.getTranslations().entrySet()
+            .stream()
+            .forEach(englishPhrase -> processEachPhrase(englishPhrase, currentUser));
+    }
+
+    private void processEachPhrase(Map.Entry<String, String> currentPhrase, UserInfo currentUser) {
+        val result = dictionaryRepository.findByEnglishPhrase(currentPhrase.getKey());
+        if (result.isPresent()) {
+            updatePhrase(currentPhrase, currentUser, result.get());
+        } else {
+            createNewPhrase(currentPhrase, currentUser);
+        }
+    }
+
+    private void createNewPhrase(Map.Entry<String, String> currentPhrase, UserInfo currentUser) {
+
+        val newEntity = isCurrentRole(currentUser)
+            ? dictionaryMapper.modelToEntityWithTranslationUploadEntity(currentPhrase, currentUser.getUid())
+            : dictionaryMapper.modelToEntityWithoutTranslationPhrase(currentPhrase);
+        dictionaryRepository.save(newEntity);
+    }
+
+    private void updatePhrase(Map.Entry<String, String> currentPhrase,
+                              UserInfo currentUser, DictionaryEntity dictionaryEntity) {
+        if (isCurrentRole(currentUser)) {
+            val translationUploadEntity = new TranslationUploadEntity();
+            translationUploadEntity.setUserId(currentUser.getUid());
+            translationUploadEntity.setUploaded(LocalDateTime.now());
+            dictionaryEntity.setTranslationPhrase(currentPhrase.getValue());
+            dictionaryEntity.setTranslationUpload(translationUploadEntity);
+            dictionaryRepository.save(dictionaryEntity);
+        }
+    }
+
+    private boolean isCurrentRole(UserInfo currentUser) {
+        return securityUtils.hasManageTranslationsRole(currentUser.getRoles(), MANAGE_TRANSLATIONS_ROLE);
```

Comment:
You will see when the conflicts are resolved, but there is now a new more generic method that has been introduced for checking the current user has a role 🙌